evals/elsuite/audio/eval.py

```diff
@@ -1,36 +1,40 @@
-import base64
+import json
 import logging
 import string
 from collections import Counter
-from io import BytesIO
-from typing import Any, List, Optional
-from urllib.parse import parse_qs, urlparse
+from typing import Any, Dict, List, Optional, Union
 
 import jiwer
-import soundfile as sf
-from datasets import Audio, load_dataset
+from datasets import Audio
 from pydantic import BaseModel
 from sacrebleu.metrics.bleu import BLEU
 
 import evals
 import evals.metrics
 from evals.api import CompletionFn
+from evals.elsuite.audio.utils import (
+    AUDIO_PLACEHOLDER,
+    DEFAULT_SAMPLE_RATE,
+    build_messages,
+    load_hf_dataset,
+    make_audio_content,
+    redact_audio_content,
+)
 from evals.elsuite.modelgraded.classify_utils import classify
 from evals.record import RecorderBase
 
 logger = logging.getLogger(__name__)
-DEFAULT_SAMPLE_RATE = 16000
-AUDIO_PLACEHOLDER = "<|audio|>"
 
 
 class Sample(BaseModel):
-    audio: Any
-    transcript: str
-    expected: str
-    context: Optional[str] = None
+    # This will typically correspond to a row in a HF dataset, although
+    # in some cases a row may produce multiple samples.
+    data: dict[str, Any]
```

Comment:
nit: is there a point to having a wrapper class for this?

Reply:
Good point, I originally was going to have other fields but then realized I didn't need them. Let me see what can be done here.